From 8d00194b9be5e97482360bc44540323fb5056051 Mon Sep 17 00:00:00 2001
From: Olivier Moysan <olivier.moysan@foss.st.com>
Date: Wed, 13 Mar 2024 11:53:10 +0100
Subject: [PATCH 1017/1141] arm64: dts: st: change mdf sitf frequency on
 stm32mp257f-dk

The mdf decimation ratio can be expressed as Fsitf / Fs where
Fsitf is the frequency of the serial interface wired to digital
microphones, and Fs the PCM sampling rate.
With Fsitf = 2048 MHz, the 48kHz sampling rate is not supported
when the reshape filter is active.
Change Fsitf to 1536MHz support 8,16, 32 and 48kHz rates with reshape
filter.
With filter order 5, the decimation ratio is out of range for 8kHz rate.
So, reduce also filter order to 4.

Signed-off-by: Olivier Moysan <olivier.moysan@foss.st.com>
Change-Id: I39b1ef613c2a24fe630f6018bd3faa2ee10b8c4e
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/366455
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-dk.dts | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
index 3552358da2d2..6630e21515f0 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
@@ -470,7 +470,7 @@ &mdf1 {
 	pinctrl-1 = <&mdf_cck0_sleep_pins_a>;
 	#clock-cells = <0>;
 	clock-output-names = "cck0";
-	clock-frequency = <2048000>;
+	clock-frequency = <1536000>;
 	status = "okay";
 
 	sitf6: sitf@380 {
@@ -483,7 +483,7 @@ sitf6: sitf@380 {
 	};
 
 	filter0: filter@84  {
-		st,cic-mode = <5>;
+		st,cic-mode = <4>;
 		st,sitf = <&sitf6 0>;
 		st,hpf-filter-cutoff-bp = <625>;
 		status = "okay";
@@ -504,7 +504,7 @@ mdf_endpoint0: endpoint {
 	};
 
 	filter1: filter@104  {
-		st,cic-mode = <5>;
+		st,cic-mode = <4>;
 		st,sitf = <&sitf6 1>;
 		st,hpf-filter-cutoff-bp = <625>;
 		status = "okay";
-- 
2.39.2

